File: app/src/main/java/com/cyberIyke/allvideodowloader/views/Badge.java
package com.cyberIyke.allvideodowloader.views;

import android.content.Context;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;
import androidx.core.content.ContextCompat;

import com.cyberIyke.allvideodowloader.R;

public class Badge extends AppCompatTextView {


    public Badge(Context context) {
        super(context);
    }

    public Badge(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public Badge(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setNumber(int number) {
        setText(String.valueOf(number));
    }

    public int getNumber(){
        return Integer.parseInt(getText().toString());
    }

    public void tabSelected(boolean isSelected) {
        if (isSelected) {
            setTextColor(ContextCompat.getColor(getContext(), R.color.white));
            setBackground(ContextCompat.getDrawable(getContext(), R.drawable.ic_tab_selected));
        } else {
            setTextColor(ContextCompat.getColor(getContext(), R.color.text_2));
            setBackground(ContextCompat.getDrawable(getContext(), R.drawable.ic_tab_unselected));
        }
    }

    public void hide() {
        setVisibility(GONE);
    }

    public void show() {
        setVisibility(VISIBLE);
    }


}

File: app/src/main/java/com/cyberIyke/allvideodowloader/views/Badge.kt
package com.cyberIyke.allvideodowloader.views

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import androidx.core.content.ContextCompat
import com.cyberIyke.allvideodowloader.R

class Badge : AppCompatTextView {
    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!, attrs
    ) {
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context!!, attrs, defStyleAttr
    ) {
    }

    var number: Int
        get() = text.toString().toInt()
        set(number) {
            text = number.toString()
        }

    fun tabSelected(isSelected: Boolean) {
        background = if (isSelected) {
            setTextColor(ContextCompat.getColor(context, R.color.white))
            ContextCompat.getDrawable(context, R.drawable.ic_tab_selected)
        } else {
            setTextColor(ContextCompat.getColor(context, R.color.text_2))
            ContextCompat.getDrawable(context, R.drawable.ic_tab_unselected)
        }
    }

    fun hide() {
        visibility = GONE
    }

    fun show() {
        visibility = VISIBLE
    }
}